fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -627,13 +627,41 @@ private ResourceDAO getResourceDao() {
         
         T resource = null;
         com.ibm.fhir.persistence.jdbc.dto.Resource resourceDTO = null;
+
+        FHIRSearchContext searchContext = context.getSearchContext();
+        List<String> elements = null;
+        //Check if _summary is required
+        if (searchContext != null && searchContext.hasSummaryParameter()) {
+            Set<String> summaryElements = null;
+            SummaryValueSet summary = searchContext.getSummaryParameter();
+            
+            switch (summary) {
+            case TRUE:
+                summaryElements = JsonSupport.getSummaryElementNames(resourceType);
+                break;
+            case TEXT:
+                summaryElements = SearchUtil.getSummaryTextElementNames(resourceType);
+                break;
+            case DATA:
+                summaryElements = JsonSupport.getSummaryDataElementNames(resourceType);
+                break;
+            default:
+                break;
+                
+            }
+
+            if (summaryElements != null) {
+                elements = new ArrayList<String>();
+                elements.addAll(summaryElements);
```

Comment:
maybe directly add elements.addAll during the case in the switchStatement?

Reply:
I want to keep element as null instead of an empty ArrayList if there is no _summary.

Reply:
right, that's the default case.

Reply:
do you mean to add these following in each case:
                elements = new ArrayList<String>();
                elements.addAll(JsonSupport.getSummaryxxxxNames(resourceType));
?

Reply:
elements = new ArrayList(); this would be before the case. 
then you would call elements.AddAll once-line per each

Reply:
I want a null elements when  _summary=false etc instead of an empty Arraylist, and null is used in filtering code logic ...   this was why I didn't  do "elements = new ArrayList()" right before the switch ...

Reply:
ah! OK. Then I approve.